examples/c/ex_all.c

```diff
@@ -376,7 +376,8 @@ cursor_ops(WT_SESSION *session)
         error_check(
           session->open_cursor(session, "table:mytable", NULL, "overwrite=false", &cursor));
         cursor->set_key(cursor, key);
-        error_check(cursor->remove(cursor));
+        if (cursor->search(cursor) == 0)
+            error_check(cursor->remove(cursor));
```

Comment:
Not sure why this is necessary, can't we just assume we're handling `notfound` in `error_check`?

Reply:
I'm not sure that I'm understanding your point correctly, but I don't think `error_check` will handle the `notfound` error code. I'm not sure if there are other situations where returning a `notfound` code is not ok. Should we create a new similar `error_check` function that allows for a `notfound` code to be returned?

Reply:
Sorry, @mm-ng, I should have been more clear. What I was trying to say is that in example code for the remove method, I would avoid including a search call, I think that complicates the example. I might be wrong about that, though!

Reply:
Right yeah, that makes sense. My intention was that we should only be removing things that exist in the database, hence the search call but yes, it might be better to check for a return code instead.

Reply:
I will change the test so that we still print a message if we hit a `notfound` but the test program will not crash as returning a `notfound` is ok and expected when the key does not exist.